libsignal-service/src/service_address.rs

```diff
@@ -1,124 +1,64 @@
-use phonenumber::*;
+use std::{convert::TryFrom, fmt};
+
 use serde::{Deserialize, Serialize};
 use uuid::Uuid;
 
-use crate::{push_service::ServiceError, session_store::SessionStoreExt};
-
-#[derive(thiserror::Error, Debug)]
+#[derive(thiserror::Error, Debug, Clone)]
 pub enum ParseServiceAddressError {
-    #[error("Supplied phone number could not be parsed in E164 format")]
-    InvalidPhoneNumber(#[from] phonenumber::ParseError),
-
-    #[error("Supplied uuid could not be parsed")]
-    InvalidUuidError(#[from] uuid::Error),
+    #[error("Supplied UUID could not be parsed")]
+    InvalidUuid(#[from] uuid::Error),
 
-    #[error("Envelope with neither Uuid or E164")]
-    NoSenderError,
+    #[error("Envelope with no Uuid")]
+    NoUuid,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct ServiceAddress {
-    pub uuid: Option<Uuid>,
-    pub phonenumber: Option<PhoneNumber>,
+    pub uuid: Uuid,
 }
 
-impl ServiceAddress {
-    /// Formats the phone number, if present, as E164
-    pub fn e164(&self) -> Option<String> {
-        self.phonenumber
-            .as_ref()
-            .map(|pn| pn.format().mode(phonenumber::Mode::E164).to_string())
-    }
-
-    pub async fn sub_device_sessions<S: SessionStoreExt>(
-        &self,
-        session_store: &S,
-    ) -> Result<Vec<u32>, ServiceError> {
-        let mut sub_device_sessions = Vec::new();
-        if let Some(uuid) = &self.uuid {
-            sub_device_sessions.extend(
-                session_store
-                    .get_sub_device_sessions(&uuid.to_string())
-                    .await?,
-            );
-        }
-        if let Some(e164) = &self.e164() {
-            sub_device_sessions
-                .extend(session_store.get_sub_device_sessions(e164).await?);
-        }
-        Ok(sub_device_sessions)
+impl fmt::Display for ServiceAddress {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.uuid)
```

Comment:
I'm not sure whether I necessarily agree with that. I think I'd rather see ServiceAddress(uuid) then.

This implies you would rely on this specific Display implementation for certain calls (see above) and that doesn't sound very healthy.